### ISSUE TITLE
types of `@typescript-eslint/eslint-plugin` are not published

### ISSUE THREAD

**Issue by JounQin** [typescript-eslint/typescript-eslint#3129]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

https://unpkg.com/browse/@typescript-eslint/eslint-plugin@4.16.1/

```TS
 import { rules as typescriptEslintRules } from '@typescript-eslint/eslint-plugin';
```

**Expected Result**

**Actual Result**

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.2` |
| `@typescript-eslint/parser`        | `4.15.2` |
| `TypeScript`                       | `4.1.3` |

**Comment by bradzacher** [comment]
We don't publish them on purpose because there is little use to them - they are only useful specifically in the case where you are extending our rules (which is literally maybe used by 0.00000001% of our users).

Additionally turning on type declarations for the package is a pain as it causes TS to enforce requirements about private types - meaning files have to export extra cruft.
Things may have changed since we first looked at it, but at the time ~2y ago - it wasn't worth the effort at all.

**Comment by JounQin** [comment]
How about add a minimal declaration file?

My temporary workaround:

```ts
declare module '@typescript-eslint/eslint-plugin' {
  import { Rule } from 'eslint'
  export const rules: Record<string, Rule.RuleModule>
  // configs can be exported too
}
```

**Comment by bradzacher** [comment]
we purposely don't have any dependency on the ESLint types as they are wrong with respect to our project, so they would have to use the types from our packages.
but otherwise, sure - a hard-coded decl file wouldn't be a bad idea necessarily.

**Comment by JounQin** [comment]
So should I PR to this repo or DefinitelyTyped instead?

**Comment by bradzacher** [comment]
Feel free to PR it here
